File: apps/dt_web/web/models/partition_scenario.ex
defmodule DtWeb.PartitionScenario do
  use DtWeb.Web, :model

  @derive {Poison.Encoder, only: [:id, :mode, :partition_id, :scenario_id]}

  schema "partitions_scenarios" do
    field :mode, :string
    belongs_to :partition, Partition
    belongs_to :scenario, Sensor

    timestamps()
  end

  def create_changeset(struct, params \\ %{}) do
    struct
    |> cast(params, [:mode, :partition_id, :scenario_id])
    |> validate_required([:mode, :partition_id, :scenario_id])
  end

  def update_changeset(struct, params \\ %{}) do
    struct
    |> cast(params, [:mode, :partition_id, :scenario_id])
    |> validate_required([:mode, :partition_id, :scenario_id])
  end

end

File: apps/dt_web/webapp/src/app/models/partitionScenario.ts
export class PartitionScenario {
  id: number;
  mode: string;
  scenario_id: number; // tslint:disable-line
  partition_id: number; // tslint:disable-line

  constructor(scenarioId) {
    this.scenario_id = scenarioId; // tslint:disable-line
  };
}
